[FLINK-30989][table] Fix some options don't take effect in batch mode

lsyldliu opened the pull request:
<!--
*Thank you very much for contributing to Apache Flink - we are happy that you want to help us improve Flink. To help the community review your contribution in the best possible way, please go through the checklist below, which will get the contribution into a shape in which it can be best reviewed.*

*Please understand that we do not do this to make contributions to Flink a hassle. In order to uphold a high standard of quality for code contributions, while at the same time managing a large number of contributions, we need contributors to prepare the contributions well, and give reviewers enough contextual information for the review. Please also understand that contributions that do not follow this guide will take longer to review and thus typically be picked up with lower priority by the community.*

## Contribution Checklist

  - Make sure that the pull request corresponds to a [JIRA issue](https://issues.apache.org/jira/projects/FLINK/issues). Exceptions are made for typos in JavaDoc or documentation files, which need no JIRA issue.
  
  - Name the pull request in the form "[FLINK-XXXX] [component] Title of the pull request", where *FLINK-XXXX* should be replaced by the actual issue number. Skip *component* if you are unsure about which is the best component.
  Typo fixes that have no associated JIRA issue should be named following this pattern: `[hotfix] [docs] Fix typo in event time introduction` or `[hotfix] [javadocs] Expand JavaDoc for PuncuatedWatermarkGenerator`.

  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
  
  - Make sure that the change passes the automated tests, i.e., `mvn clean verify` passes. You can set up Azure Pipelines CI ## What is the purpose of the change

*Currently, some table execution options don't pass to runtime correctly, causing them don't take effect when users configure it, so we should fix it.*


## Brief change log

  - *Fix some table execution options that don't pass to runtime correctly*


## Verifying this change

This change is already covered by existing tests.


## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? (not documented)

lsyldliu commented:
cc @godfreyhe

godfreyhe commented on flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/batch/BatchExecHashJoin.java:
```diff
@@ -213,6 +213,12 @@ protected Transformation<RowData> translateToPlanInternal(
                         condFunc,
                         1.0 * externalBufferMemory / managedMemory);
 
+        boolean compressionEnable =
```
nit: compressionEnabled

godfreyhe commented on flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/utils/SorMergeJoinOperatorUtil.java:
```diff
@@ -48,6 +49,17 @@ public static SortMergeJoinFunction getSortMergeJoinFunction(
 
+        int maxNumFileHandles =
+                config.get(ExecutionConfigOptions.TABLE_EXEC_SORT_MAX_NUM_FILE_HANDLES);
+        boolean compressionEnable =
```
compressionEnabled

godfreyhe commented on flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/utils/SorMergeJoinOperatorUtil.java:
```diff
@@ -48,6 +49,17 @@ public static SortMergeJoinFunction getSortMergeJoinFunction(
+                (int)
+                        config.get(ExecutionConfigOptions.TABLE_EXEC_SPILL_COMPRESSION_BLOCK_SIZE)
+                                .getBytes();
+        boolean asyncMergeEnable =
```
asyncMergeEnabled

godfreyhe reviewed (commented):
Thanks for the contribution, LGTM overall, I left some minor comments. btw, it's better we can also remove the config from `GeneratedClass`

lsyldliu commented:
@godfreyhe Thanks for reviewing, I've addressed your comments. Regarding removing the config from `GeneratedClass`, I think we can complete it in another PR.

godfreyhe reviewed (approved):
LGTM
